src/otx/algo/detection/rtdetr.py

```diff
@@ -0,0 +1,438 @@
+        original_size: tuple[int, int] | None = None,
+        deploy_mode: bool = False,
+    ) -> dict[str, Tensor] | tuple[list[Tensor], list[Tensor], list[Tensor]]:
+        """Post-processes the model outputs.
+
+        Args:
+            outputs (dict[str, Tensor]): The model outputs.
+            original_size (tuple[int, int], optional): The original size of the input images. Defaults to None.
+            deploy_mode (bool, optional): Whether to run in deploy mode. Defaults to False.
+
+        Returns:
+            dict[str, Tensor] | tuple[list[Tensor], list[Tensor], list[Tensor]]: The post-processed outputs.
+        """
+        logits, boxes = outputs["pred_logits"], outputs["pred_boxes"]
+
+        # convert bbox to xyxy and rescale back to original size (resize in OTX)
+        bbox_pred = torchvision.ops.box_convert(boxes, in_fmt="cxcywh", out_fmt="xyxy")
+        if not deploy_mode and original_size is not None:
+            original_size_tensor = torch.tensor(original_size).to(bbox_pred.device)
+            bbox_pred *= original_size_tensor.repeat(1, 2).unsqueeze(1)
+
+        # perform scores computation and gather topk results
+        scores = nn.functional.sigmoid(logits)
+        scores, index = torch.topk(scores.flatten(1), self.num_top_queries, axis=-1)
+        labels = index % self.num_classes
+        index = index // self.num_classes
+        boxes = bbox_pred.gather(dim=1, index=index.unsqueeze(-1).repeat(1, 1, bbox_pred.shape[-1]))
+
+        if deploy_mode:
+            return {"bboxes": boxes, "labels": labels, "scores": scores}
+
+        scores_list, boxes_list, labels_list = [], [], []
+
+        for sc, bb, ll in zip(scores, boxes, labels):
+            scores_list.append(sc)
+            boxes_list.append(
+                torchvision.tv_tensors.BoundingBoxes(bb, format="xyxy", canvas_size=original_size),
+            )
+            labels_list.append(ll.long())
+
+        return scores_list, boxes_list, labels_list
+
+    def export(
+        self,
+        batch_inputs: Tensor,
+        batch_img_metas: list[dict],
+        explain_mode: bool = False,
+    ) -> dict[str, Any] | tuple[list[Any], list[Any], list[Any]]:
+        """Exports the model."""
+        if explain_mode:
+            msg = "Explain mode is not supported for RTDETR model yet."
+            raise NotImplementedError(msg)
+        return self.postprocess(self._forward_features(batch_inputs), deploy_mode=True)
+
+
+class OTXRtDetr(ExplainableOTXDetModel):
+    image_size = (1, 3, 640, 640)
+    mean = (0.0, 0.0, 0.0)
+    std = (255.0, 255.0, 255.0)
+
+    def _customize_inputs(
+        self,
+        entity: DetBatchDataEntity,
+        pad_size_divisor: int = 32,
+        pad_value: int = 0,
+    ) -> dict[str, Any]:
+        return {
+            "images": entity.images,
+            "targets": [{"boxes": bb, "labels": ll} for bb, ll in zip(entity.bboxes, entity.labels)],
+        }
+
+    def _customize_outputs(
```

Comment:
There doesn't seem to be any unit-test cover for this function, shouldn't it be added?

Reply:
I am not sure if it is needed, since it is pretty simple, but I can add it of course

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/3741/commits/2d5cef1ef2bbc576bebd10fd1a96e812d6ef0548